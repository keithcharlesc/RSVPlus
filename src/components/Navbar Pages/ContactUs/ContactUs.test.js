import React from "react";
import MockContactUs from "./MockContactUs";
import { render, getByTestId } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import { toHaveTextContent } from "@testing-library/jest-dom";

test("renders without crashing", () => {
  const { getByText } = render(<MockContactUs />);
  getByText("Form");
  getByText("Name");
  getByText("Email");
  getByText("Message");
  expect(
    getByTestId(document.documentElement, "submit-button")
  ).toHaveTextContent("Send Message!");
});

it("matches snapshot", () => {
  const tree = TestRenderer.create(<MockContactUs />).toJSON();
  expect(tree).toMatchSnapshot();
});
