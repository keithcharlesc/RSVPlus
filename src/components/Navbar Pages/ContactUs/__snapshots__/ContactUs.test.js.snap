// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div>
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        className="nav-bar-appName text-danger navbar-brand"
        href="/"
      >
        RSVP+
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/"
            href="/"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Home
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/channels"
            href="/channels"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Channels
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/calendar"
            href="/calendar"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Calendar
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/faq"
            href="/faq"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            FAQ
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/contact-us"
            href="/contact-us"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Contact Us
          </a>
        </div>
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <span
            className="nav-bar-name navbar-text"
          >
            Signed in as: 
            Bob
            <img
              className="photo ml-2 rounded-circle"
              src="thttp://email.com/image"
            />
          </span>
        </div>
      </div>
      <button
        className="ml-3 btn btn-outline-primary"
        data-testid="button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Log Out
      </button>
    </nav>
    <div>
      
    </div>
  </div>,
  <div
    className="p-3 mb-2 bg-dark text-white"
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <h2
      className="page-header text-center"
    >
      CONTACT US
    </h2>
    <div
      className="d-flex align-items-center justify-content-center container"
      style={
        Object {
          "minHeight": "75vh",
        }
      }
    >
      <div
        className="mt-5 card"
      >
        <div
          className="card-body"
        >
          <h3
            className="text-center mb-4"
          >
            Form
          </h3>
          
          
          <form
            className="text-dark"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
              id="name"
            >
              <label
                className="form-label"
              >
                Name
              </label>
              <input
                className="form-control"
                required={true}
                type="name"
              />
            </div>
            <br />
            <div
              className="form-group"
              id="email"
            >
              <label
                className="form-label"
              >
                Email
              </label>
              <input
                className="form-control"
                required={true}
                type="email"
              />
            </div>
            <br />
            <div
              className="form-group"
              id="message"
            >
              <label
                className="form-label"
              >
                Message
              </label>
              <input
                className="form-control"
                required={true}
                type="message"
              />
            </div>
            <br />
            <button
              className="w-100 btn btn-danger"
              data-testid="submit-button"
              disabled={false}
              type="submit"
            >
              Send Message!
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
]
`;
